File: app/src/main/java/my/edu/tarc/communechat_v2/LocalDatabase/Converters.java
package my.edu.tarc.communechat_v2.LocalDatabase;

import android.arch.persistence.room.TypeConverter;

import com.google.android.gms.common.util.NumberUtils;

import java.util.Calendar;
import java.util.GregorianCalendar;

public class Converters {
    @TypeConverter
    public static Calendar calendarFromTimestamp(String value) {
        if (value == null) {
            return null;
        }
        Calendar cal = new GregorianCalendar();
        cal.setTimeInMillis(NumberUtils.parseHexLong(value) * 1000);
        return cal;
    }

    @TypeConverter
    public static String dateToTimestamp(Calendar cal) {
        if (cal == null) {
            return null;
        }
        return "" + cal.getTimeInMillis() / 1000;
    }
}

File: app/src/main/java/my/edu/tarc/communechat_v2/LocalDatabase/Converters.kt
package my.edu.tarc.communechat_v2.LocalDatabase

import android.arch.persistence.room.TypeConverter
import com.google.android.gms.common.util.NumberUtils
import java.util.*

object Converters {
    @TypeConverter
    fun calendarFromTimestamp(value: String?): Calendar? {
        if (value == null) {
            return null
        }
        val cal = GregorianCalendar()
        cal.timeInMillis = NumberUtils.parseHexLong(value) * 1000
        return cal
    }

    @TypeConverter
    fun dateToTimestamp(cal: Calendar?): String? {
        return if (cal == null) {
            null
        } else "" + cal.timeInMillis / 1000
    }
}
